source/go-perun/multi_ledger_tutorial/index.rst

```diff
@@ -0,0 +1,160 @@
+   :language: go
+   :lines: 62-68
+
+The initialization of the Perun client is similar as in the payment channel tutorial,
+however, here we will highlight the differences regarding the multi-ledger functionality.
+For a more detailed explanation, we refer to the :ref:`payment client section <payment_client>`.
+
+First of all, we will use special variants of the funder and adjudicator which are located in the ``multi`` package of
+go-perun. Essentially, the multi-ledger funder and multi-ledger adjudicator contain a funder or adjudicator for each
+ledger identified by its ID. Just for curiosity, we can have a brief look how the multi-ledger funder is actually
+implemented in go-perun:
+
+.. literalinclude:: ../source/go-perun/channel/multi/funder.go
+   :caption: `👇 This code on GitHub. <https://github.com/hyperledger-labs/go-perun/blob/e831f7ec3d1fee283cc66a3035969a2c7b9aad71/channel/multi/funder.go#L26>`__
+   :language: go
+   :lines: 26-40
+
+Having this in mind, we can jump back to our code and see how we set up the multi funder & adjudicator (note the
+highlighted lines). After we have initialized them, we iterate over the chains array and proceed with creating the
+contract backend for this chain, validating the contracts, and create a traditional funder & adjudicator to
+register them on the multi-ledger variants.
+
+.. literalinclude:: ../../perun-examples/multiledger-channel/client/client.go
+   :emphasize-lines: 3,4,33,38
+   :caption: `👇 This code on GitHub. <https://github.com/hyperledger-labs/go-perun/blob/e831f7ec3d1fee283cc66a3035969a2c7b9aad71/channel/multi/funder.go#L26>`__
+   :language: go
+   :lines: 69-107
+
+After that, we only need to set up a watcher for our client who is watching for events on our multi-ledger adjudicator,
+and we have all components ready to create the Perun client by using ``client.New``.
+
+.. literalinclude:: ../../perun-examples/multiledger-channel/client/client.go
+   :emphasize-lines: 9
+   :caption: `👇 This code on GitHub. <https://github.com/hyperledger-labs/go-perun/blob/e831f7ec3d1fee283cc66a3035969a2c7b9aad71/channel/multi/funder.go#L26>`__
+   :language: go
+   :lines: 110-121
+
+Finally, we can create our Payment client which gets the Perun client, and then start the handler of the Perun client to
+handle channel and update proposals.
+
+.. literalinclude:: ../../perun-examples/multiledger-channel/client/client.go
+   :caption: `👇 This code on GitHub. <https://github.com/hyperledger-labs/go-perun/blob/e831f7ec3d1fee283cc66a3035969a2c7b9aad71/channel/multi/funder.go#L26>`__
+   :language: go
+   :lines: 123-130
+
+Channel opening
+...............
+
+The channel opening is again very similar to the payment channel tutorial, the only difference is visible
+when setting the allocations of the channel proposal.
+There, we now have two assets (the ERC20 tokens) for the two chains for which we need to define the balances for Alice
```

Comment:
Why do you use ERC20 tokens and not just basic ETH? (would simplify the tutorial)

Reply:
Because I think it's more "exciting" to do a cross-chain swap with ERC20 tokens instead of two times ETH.
Also, we can showcase that go-perun (resp. the Ethereum backend) also supports ERC20 tokens with this tutorial.

Reply:
The point of the tutorials should usually be to keep it as simple as possible. We can have an ERC20 tutorial separately.

Reply:
> The point of the tutorials should usually be to keep it as simple as possible. We can have an ERC20 tutorial separately.
In my opinion, it does not make it too complicated when using ERC20 tokens for the tutorial instead of ETH, so I think it's worth it. Also, the examples repo is alreay merged.